File: New folder/notes.fs
// fsharp notes
let greet name = sprintf "hello %s" name

[<EntryPoint>]
let main argv =
    printfn "%s" (greet "practice")
    let nums = [1; 2; 3]
    let doubled = nums |> List.map (fun n -> n * 2)
    printfn "%A" doubled
    0
// TODO: try discriminated unions next

File: New folder/notes.zig
// zig notes
const std = @import("std");

fn greet(name: []const u8) void {
    std.debug.print("hello {s}\n", .{name});
}

pub fn main() void {
    greet("practice");

    const nums = [_]i32{1, 2, 3};
    for (nums) |n| {
        std.debug.print("{}\n", .{n * 2});
    }
}
// TODO: try allocators next
